presto-parquet/src/main/java/io/prestosql/parquet/reader/ParquetReader.java

```diff
@@ -122,6 +129,16 @@ public ParquetReader(
         chunkReaders = dataSource.planRead(ranges);
     }
 
+    public MessageType getFileSchema()
+    {
+        return fileSchema;
+    }
+
+    public MessageColumnIO getMessageColumn()
+    {
+        return this.messageColumn;
+    }
+
```

Comment:
These are unnecessary. They are there only so that `createPageSource` can get them out of the reader. If, instead, you inlined the `createPageSource` method or structured it differently, you could use them directly in the caller.